int32 vel
int32 trq

